test/unified_memory_allocation/memoryPool.hpp

```diff
@@ -52,4 +74,42 @@ TEST_P(umaPoolTest, pow2AlignedAlloc) {
     }
 }
 
+// TODO: add similar tests for realloc/aligned_alloc, etc.
+// TODO: add multithreaded tests
+TEST_P(umaMultiPoolTest, memoryTracking) {
+    static constexpr int allocSizes[] = {1,  2,  4,  8,   16,   20,
+                                         32, 40, 64, 128, 1024, 4096};
+    static constexpr auto nAllocs = 256;
+
+    std::mt19937_64 g(0); // TODO: test with random seeds?
```

Comment:
no, tests should always use a fixed seed.

Reply:
I strongly dislike when CI tests randomly fail because of some non-determinism in a test. If you need something with a lot of cases, I'd rather we do proper fuzzing in a separate job.